### ISSUE TITLE
Implement blockdata consumer endpoint

### ISSUE THREAD

**Issue by tsani** [labcoders/atuin-server#2]

**Comment by kevin-li-195** [comment]
Am I correct in thinking that we won't have to ever pull the world into CC? We'll only ever have to query the DB from the server and perform all computation there?

**Comment by tsani** [comment]
That's correct. Turtles might transiently hold part of the world state, but the world state will by and large be held in the DB (which is tricky to serialize now since it's in Postgres).

For example, pathfinding will be done on the server. Once the path is computed, the server will lock the blocks along the path, perform Lua codegen to make a program that the turtle can run, and will ship the program to the turtle. No world state is ever held on the turtle.
